SignalMessaging/utils/LRUCache.swift

```diff
@@ -21,39 +21,61 @@ public class AnyLRUCache: NSObject {
     public func set(key: NSObject, value: NSObject) {
         self.backingCache.set(key: key, value: value)
     }
+
+    @objc
+    public func clear() {
+        self.backingCache.clear()
+    }
 }
 
 // A simple LRU cache bounded by the number of entries.
-//
-// TODO: We might want to observe memory pressure notifications.
 public class LRUCache<KeyType: Hashable & Equatable, ValueType> {
 
     private var cacheMap: [KeyType: ValueType] = [:]
     private var cacheOrder: [KeyType] = []
     private let maxSize: Int
 
+    @objc
     public init(maxSize: Int) {
         self.maxSize = maxSize
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(didReceiveMemoryWarning),
```

Comment:
I think it's cool to make are LRUCache fancier like this, but I still think a lot of work has gone into NSCache, and expect that on average we'll get better performance by using it.